// @whcg/whcg-lit-grid/dist/index.es6.js
import '@whcg/whcg-lit-grid/theme/whcg-lit-grid-styles.js';
import '@vaadin/vaadin-grid/src/vaadin-grid.js';
import "@vaadin/vaadin-grid/src/vaadin-grid-selection-column.js";
// import "@vaadin/vaadin-grid/vaadin-grid-column.js";
// import "@vaadin/vaadin-grid/vaadin-grid-sorter.js";
// import "@vaadin/vaadin-grid/vaadin-grid-filter.js";
import '@polymer/iron-ajax';
import '@polymer/polymer/lib/elements/dom-bind.js';
import { html, PolymerElement } from '@polymer/polymer/polymer-element.js';

class WhcgLitGrid extends PolymerElement {
    constructor() {
        super();
        //this.db = new PouchDB('http://localhost:5984/statistics');
        // this.selectedItems = [];
        this.tableheight = 400;
    }

    static get properties() {
        return {
            items: {
                type: Array,
                readOnly: false,
                notify: false,
                observer: '_itemChanged'
            },
            selecteditems: {
                type: Array,
                readOnly: false,
                notify: true,
                observer: '_selecteditemsChanged'
            },
            pageSize: {
                type: Number,
                readOnly: false,
                notify: true,
                observer: '_pagesizeChanged'
            },
            tableheight: {
                type: Number,
                readOnly: false,
                notify: true,
            }
        }
    }

    connectedCallback() {
        super.connectedCallback();
        // let that = this;
        // this.db.find({
        //     //selector: { raw_real_estate_company_name: { $gte: "a" } },
        //     //selector: { municipality: { $eq: "NACKA" } },
        //     // selector: {
        //     //   municipality: {$or: [
        //     //     { $eq: 'STOCKHOLM' },
        //     //     { $eq: 'NACKA' }
        //     //   ]}
        //     // },
        //     selector: { 
        //       raw_real_estate_company_name: { $gte: "a" },
        //       municipality: { $gte: null },
        //       build_year: { $gte: "1955" }
        //     },
        //     limit: 10
        // }).then(function (result) {
        //     console.log(result);
        //     console.log(result.docs);
        //    // that.items = result.docs;
        // }).catch(function (err) {
        //     console.log(err);
        // });
    }


    _itemChanged() {
        // console.log('_itemChanged');
    }

    _selecteditemsChanged() {
        // console.log('_selecteditemsChanged');
        // console.log(this.selecteditems)
    }

    _gridChanged(e) {
        console.log('_gridChanged');
        console.log(this.$.grid.selectedItems);
        //let newSelection = this.$.grid.selectedItems;
        let newSelection = this.$.grid.selectedItems.slice();
        // this.selecteditems = [];
        this.selecteditems = newSelection;
    }

    static get template() {
        return html`
        <style>

            :root {
                background-color: transparent;
            }


            

        </style>


                <vaadin-grid id="grid" aria-label="Basic Binding Example" items="[[items]]" theme="wrap-cell-content" on-change="_gridChanged" style$="height: {{tableheight}}px">
                    <slot></slot>
                </vaadin-grid>




    `;
    };   
}

window.customElements.define('whcg-lit-grid', WhcgLitGrid);

export { WhcgLitGrid };




//HEJ
